src/components/asset-list/RecyclerAssetList2/FastComponents/FastBalanceCoinRow.tsx

```diff
@@ -0,0 +1,330 @@
+import React, { useCallback, useRef } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { CoinIconIndicator } from '../../../../components/coin-icon';
+import { Icon } from '../../../../components/icons';
+import { ButtonPressAnimation } from '../../../animations';
+
+import { initialChartExpandedStateSheetHeight } from '../../../expanded-state/asset/ChartExpandedState';
+import { ExtendedState } from '../core/RawRecyclerList';
+import FastCoinIcon from './FastCoinIcon';
+import { Text } from '@rainbow-me/design-system';
+import {
+  useAccountAsset,
+  useCoinListFinishEditingOptions,
+} from '@rainbow-me/hooks';
+import Routes from '@rainbow-me/routes';
+import { borders, colors, padding, shadow } from '@rainbow-me/styles';
+
+interface CoinCheckButtonProps {
+  isHidden: boolean;
+  isPinned: boolean;
+  onPress: () => void;
+  uniqueId: string;
+  theme: any;
+}
+
+const CoinCheckButton = React.memo(function CoinCheckButton({
+  isHidden,
+  isPinned,
+  onPress,
+  uniqueId,
+  theme,
+}: CoinCheckButtonProps) {
+  const { selectedItems } = useCoinListFinishEditingOptions();
+  const selected = selectedItems.includes(uniqueId);
+
+  const showOutline = !(isHidden || isPinned);
+  const coinIconPlaceholder = !selected && (isHidden || isPinned);
+
+  const checkmarkBackgroundDynamicStyle = {
+    ...shadow.buildAsObject(
+      0,
+      4,
+      12,
+      theme.isDarkMode ? colors.shadow : colors.appleBlue,
+      0.4
+    ),
+  };
+
+  return (
+    <View style={cx.checkboxContainer}>
+      <ButtonPressAnimation onPress={onPress}>
+        <View style={cx.checkboxInnerContainer}>
+          {showOutline && (
+            <View
+              style={[
+                cx.circleOutline,
+                { borderColor: colors.alpha(colors.blueGreyDark, 0.12) },
+              ]}
+            />
+          )}
+
+          {coinIconPlaceholder && (
+            <CoinIconIndicator
+              // @ts-expect-error
```

Comment:
For whatever reason, it didn't work because of the way we define that component

```js
// doesn't work
const ComponentName = React.memo(function ComponentName({ isPinned }) {});
export default ComponentName;

// works
function ComponentName({ isPinned }) {};
export default React.memo(ComponentName);
```